x-pack/plugins/apm/public/components/app/Settings/anomaly_detection/add_environments.tsx

```diff
@@ -44,8 +44,9 @@ export const AddEnvironments = ({
     { preservePreviousData: false }
   );
 
-  const availableEnvironmentOptions = data.map((env) => ({
-    label: env,
+  const environments = [...data, ENVIRONMENT_NOT_DEFINED];
```

Comment:
Preferably `ENVIRONMENT_NOT_DEFINED` should only be added if some services don't have environments.
Would it be possible to add this with `missing: ENVIRONMENT_NOT_DEFINED` in the terms aggs? (double check how this affects agent config)

Reply:
this creates a new option in agent configuration, which didn't exist before:
![Screen Shot 2020-07-06 at 10 59 11 AM](https://user-images.githubusercontent.com/1967266/86624255-da239580-bf77-11ea-88e0-83a6a38fd490.png)
I could update the label to `Not defined` like in other places. The result is that when apm server requests for settings, it will include settings for the `ENVIRONMENT_NOT_DEFINED` environment. If we're OK with that in this PR, then i can ship it as part of this.

Reply:
apm server would need to request it with ENVIRONMENT_NOT_DEFINED as the specified environment:
```
$ echo '{"service":{"name": "opbeans-dotnet", "environment": "ENVIRONMENT_NOT_DEFINED"}}' |  http -a 'user:passwd' POST :5603/foo/api/apm/settings/agent-configuration/search kbn-xsrf:true
{
    ...
    "_source": {
        "@timestamp": 1594061630230,
        "agent_name": "dotNet",
        "applied_by_agent": false,
        "etag": "3368a2a0fd74cf29e234ec0614211c6af458c794",
        "service": {
            "environment": "ENVIRONMENT_NOT_DEFINED",
            "name": "opbeans-dotnet"
        },
        "settings": {
            "transaction_sample_rate": ".1"
        }
    }
}
```
as opposed to how they might get it now, by omitting the `service.environment` in the POST data:
```
echo '{"service":{"name": "opbeans-dotnet"}}' |  http -a 'user:passwd' POST :5603/foo/api/apm/settings/agent-configuration/search kbn-xsrf:true
{
    ...
    "_source": {
        "@timestamp": 1594062874882,
        "applied_by_agent": true,
        "etag": "c8b485e47a65b4230b77570930cbffa4cd8c5aa8",
        "service": {},
        "settings": {
            "transaction_sample_rate": ".75"
        }
    }
}
```
Since the latter is also how apm server get the the `All` evironment configuration, there's some ambiguity that it (or the agent) would need to resolve.

Reply:
we can still add this missing value if we pass in an option to `getAllEnvironments` like `{includeMissing: boolean}` or by filtering out `ENVIRONMENT_NOT_DEFINED` from the options in agent configuration.

Reply:
We shouldn't change the behaviour for agent config. `includeMissing`  sounds like a good plan 👍